src/Components/FileTree.tsx

```diff
@@ -0,0 +1,64 @@
+import { FileEntry } from '@tauri-apps/api/fs';
+import { VscFile, VscFolder } from 'react-icons/vsc';
+
+import { cx } from '../cx';
+
+interface FileTreeBaseProps {
+  root?: boolean;
+  files: FileEntry[];
+  file?: FileEntry;
+  selectedFiles: FileEntry[];
+  onClick: (file: FileEntry) => void;
+  rightAction?: (file: FileEntry) => React.ReactNode;
+}
+export function FileTree({ files, root, ...props }: FileTreeBaseProps) {
+  return (
+    <div>
+      <ul className={root ? 'ml-4' : 'ml-6'}>
+        {files.map((file) => (
+          <FileTreeNode file={file} files={files} key={file.path} {...props} />
```

Comment:
I think you're passing `files` to the `FileTreeNode` but the component never uses it. Maybe you should create a different type for `FileTreeNodeProps`, like 
```typescript
type FileTreeNodeProps = Omit<FileTreeBaseProps, 'root' | 'files'>
```
Or the other way around, to have only interfaces:
```typescript
interface FileTreeNodeProps {
  file?: FileEntry;
  selectedFiles: FileEntry[];
  onClick: (file: FileEntry) => void;
  rightAction?: (file: FileEntry) => React.ReactNode;
}

interface FileTreeProps extends FileTreeNodeProps {
  root?: boolean;
  files: FileEntry[];
}```

Reply:
I didn't changed this as we should implement a more robust component for FileTree (including collapsible folders for  example).